### PR TITLE
[LLM] Multi-process and distributed QLoRA on CPU platform

### PR DESCRIPTION
**Pull request by Uxito-Ada**
## Description

 Multi-process and distributed QLoRA fine-tuning on SPR server enabled by oneccl, which can be used in docker and k8s respectively.

### 1. Why the change?

as above

### 2. User API changes

**1. Multi-process mode in one docker**
```bash
docker run -itd \
 --name=bigdl-llm-fintune-qlora-cpu \
 -e STANDALONE_DOCKER=TRUE \
......
```

By enable ENV `STANDALONE_DOCKER` to **TRUE** when starting a docker container, `start-qlora-finetuning-on-cpu.sh` script can get the context and turn to apply oneccl for multi-process fine-tuning illustrated as below: 
<br>

**2. Distributed mode on k8s**

Deploy by helm chart:

image

### 3. Summary of the change 

[LLM] Multi-process and distributed QLoRA on CPU platform

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

File: docker/llm/finetune/qlora/cpu/docker/bigdl-qlora-finetuing-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
```
Comment (Uxito-Ada) [inline docker/llm/finetune/qlora/cpu/docker/bigdl-qlora-finetuing-entrypoint.sh:1]: This is for k8s, should not be combined with start script, and used in helm chart

File: docker/llm/finetune/qlora/cpu/docker/Dockerfile [hunk 2]
```diff
@@ -22,18 +46,25 @@ RUN apt-get update && \
...
-RUN pip3 install --upgrade pip && \
-    export PIP_DEFAULT_TIMEOUT=100 && \
-    pip install --upgrade torch==2.0.1 --index-url https://download.pytorch.org/whl/cpu && \
-    pip install oneccl_bind_pt -f https://developer.intel.com/ipex-whl-stable && \
-    # install CPU bigdl-llm
-    pip install --pre --upgrade bigdl-llm[all] -i https://pypi.tuna.tsinghua.edu.cn/simple/ && \
-    # install huggingface dependencies
-    pip install transformers==4.34.0 && \
-    pip install peft==0.5.0 datasets
+ADD https://raw.githubusercontent.com/intel-analytics/BigDL/main/python/llm/example/CPU/QLoRA-FineTuning/alpaca-qlora/alpaca_qlora_finetuning_cpu.py /bigdl/qlora_finetuning_cpu.py
```
Comment (Uxito-Ada) [inline docker/llm/finetune/qlora/cpu/docker/Dockerfile:62]: Hi @hzjane , I changed the Dockerfile to download alpaca_qlora_finetuning_cpu.py into image, as we discussed before. Moreover, I refactored files used to be under /, to /bigdl now

File: docker/llm/finetune/qlora/cpu/docker/Dockerfile [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
+    export PIP_DEFAULT_TIMEOUT=100 && \
+    pip install --upgrade torch==2.1.0 --index-url https://download.pytorch.org/whl/cpu && \
+# install CPU bigdl-llm
+    pip3 install --pre --upgrade bigdl-llm[all] -i https://pypi.tuna.tsinghua.edu.cn/simple/ && \
+# install ipex and oneccl
+    pip install intel_extension_for_pytorch==2.0.100 && \
+    pip install oneccl_bind_pt -f https://developer.intel.com/ipex-whl-stable && \
+# install huggingface dependencies
+    pip install transformers==4.34.0 && \
+    pip install peft==0.5.0 datasets && \
```
Comment (hzjane) [inline docker/llm/finetune/qlora/cpu/docker/Dockerfile:36]: please update it refer to [this pr](https://github.com/intel-analytics/BigDL/pull/9511)

Comment (Uxito-Ada) [reply docker/llm/finetune/qlora/cpu/docker/Dockerfile]: done

**Review by hzjane — APPROVED** [review]
LGTM

**Pull request merged**